data/src/main/java/com/framgia/data/repositoryimpl/GenreRepositoryImpl.kt

```diff
@@ -0,0 +1,26 @@
+package com.framgia.data.repositoryimpl
+
+import com.framgia.data.model.GenreEntityMapper
+import com.framgia.data.remote.api.MovieApi
+import com.framgia.domain.model.Genre
+import com.framgia.domain.repository.GenreRepository
+import io.reactivex.Single
+import javax.inject.Inject
+
+/**
+ * Created: 02/08/2018
+ * By: Sang
+ * Description:
+ */
+class GenreRepositoryImpl @Inject constructor(
+    private val movieApi: MovieApi,
+    private val genreEntityMapper: GenreEntityMapper
+) : GenreRepository {
+
+    override fun getGenres(): Single<List<Genre>> =
+        movieApi.getGenres().flatMap { response ->
```

Comment:
chỗ này có thể viết như này cho gọn
movieApi.getGenres().map { mapper.mapToDomain(it) }

Reply:
cái it là response anh ạ